openfeature/provider/registry.py

```diff
@@ -1,6 +1,10 @@
 import typing
 
 from openfeature.evaluation_context import EvaluationContext
+from openfeature.event import (
+    ProviderEventDetails,
+    _provider_status_to_event,
```

Comment:
Should this function also be made public?

Reply:
We could put it as a ProviderStatus method perhaps?

Reply:
Yeah, I think a classmethod on the `ProviderStatus` class would be ideal here!

Reply:
I ended up adding a classmethod to `ProviderEvent` because I think it makes more sense to place the dependency in that direction, happy to reconsider though.